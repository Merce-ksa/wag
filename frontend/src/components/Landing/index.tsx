import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#696969'

  },
  text: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#fff',
    textAlign: 'center',
    padding: 50,
    marginTop: 50,
    backgroundColor: '#34558b'
  }
});

export default function Landing() {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Bienvenida a WAG</Text>
      <Text style={styles.text}>Marixoxo</Text>
    </View>
  );
}
