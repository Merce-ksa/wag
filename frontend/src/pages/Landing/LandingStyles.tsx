import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {
    flex: 1
  },
  welcome: {
    padding: 5,
    textAlign: 'left',
    fontSize: 20,
    color: 'black'
  },
  brand: {
    textAlign: 'left',
    padding: 5,
    fontSize: 38
  },
  description: {
    textAlign: 'left',
    fontSize: 14,
    fontWeight: 'bold',
    padding: 5,
    color: '#7B7F9E'
  },
  autenticationContainer: {
    marginTop: 50,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 10
  },
  loginText: {
    color: '#567DF4',
    fontWeight: 'bold'
  },
  register: {
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 10,
    borderRadius: 5,
    backgroundColor: '#567DF4'
  },
  registerTetx: {
    color: '#fff',
    textAlign: 'center',
    fontSize: 18,
    padding: 20
  }
})
