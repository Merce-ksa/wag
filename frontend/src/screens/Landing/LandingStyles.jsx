import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {
    flex: 1
  },
  landingMain: {
    marginTop: 40,
    paddingLeft: 20,
    paddingRight: 20
  },
  subtitle: {
    textAlign: 'left',
    fontSize: 20,
    color: 'black'
  },
  title: {
    paddingBottom: 5,
    textAlign: 'left',
    fontSize: 65
  },
  description: {
    textAlign: 'left',
    fontSize: 14,
    color: '#7B7F9E'
  },
  autenticationContainer: {
    marginTop: 50,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 10
  },
  loginText: {
    fontSize: 16,
    color: '#567DF4'
  },
  register: {
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 10,
    borderRadius: 5,
    backgroundColor: '#567DF4'
  },
  registerTetx: {
    padding: 20,
    fontFamily: 'interSemiBold',
    fontSize: 18,
    color: '#fff',
    textAlign: 'center'
  }
})
