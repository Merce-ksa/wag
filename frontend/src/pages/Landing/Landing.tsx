import React from 'react'
import { Text, View, TouchableOpacity } from 'react-native'
import LandingStyles from './LandingStyles'
import Hero from '../../components/Hero/Hero'

export default function Landing ({ navigation }: any) {
  return (
    <View style={LandingStyles.container}>
      <Hero />
      <View style={LandingStyles.welcome}>
        <Text style={LandingStyles.welcome}>Welcome to</Text>
        <Text style={LandingStyles.brand}>WAG</Text>
        <Text style={LandingStyles.description}>The platform to store and share digital material! Do not lose anything!</Text>
        <Text style={LandingStyles.description}>Join For Free</Text>
      </View>
      <View style={LandingStyles.autenticationContainer}>
        <TouchableOpacity
          testID="login"
          onPress={() =>
            navigation.navigate('Login', { name: 'Login' })
          }
        >
          <Text style = {LandingStyles.loginText}> I have an account </Text>
        </TouchableOpacity>
        <TouchableOpacity
          testID="register"
          style = {LandingStyles.register}
          onPress={() =>
            navigation.navigate('Register', { name: 'Register' })
          }
        >
          <Text style = {LandingStyles.registerTetx}> Sing up </Text>
        </TouchableOpacity>
      </View>

    </View>
  )
}
