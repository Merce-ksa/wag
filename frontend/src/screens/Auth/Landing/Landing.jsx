import React from 'react'
import { Text, View, TouchableOpacity, Image } from 'react-native'
import LandingStyles from './LandingStyles'
import PropTypes from 'prop-types'

export default function Landing ({ navigation }) {
  return (
    <View style={LandingStyles.landingContainer}>
      <View>
        <Image
          source={require('../../../assets/images/illustration.png')}
        />
      </View>
      <View style={LandingStyles.landingMain}>
        <Text style={[{ fontFamily: 'interExtraLight' }, LandingStyles.subtitle]}>Welcome to</Text>
        <Text style={[{ fontFamily: 'barlowLight' }, LandingStyles.title]}>WAG</Text>
        <Text style={[{ fontFamily: 'interSemiBold' }, LandingStyles.description]}>The platform to store and share digital material! Do not lose anything!</Text>
        <Text style={[{ fontFamily: 'interSemiBold' }, LandingStyles.description]}>Join For Free</Text>

        <View style={LandingStyles.autenticationContainer}>
          <TouchableOpacity
            testID="login"
            onPress={() =>
              navigation.navigate('Login', { name: 'Login' })
            }
          >
            <Text style={[{ fontFamily: 'interBold' }, LandingStyles.loginText]}> I have an account </Text>
          </TouchableOpacity>
          <TouchableOpacity
            testID="register"
            style = {LandingStyles.register}
            onPress={() => navigation.navigate('Register', { name: 'Register' })}
          >
            <Text style = {LandingStyles.registerTetx}> Sing up </Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  )
}

Landing.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  })
}
