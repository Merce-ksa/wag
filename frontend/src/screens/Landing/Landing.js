import React from 'react'
import { Text, View, TouchableOpacity } from 'react-native'
import AppLoading from 'expo-app-loading'
import { useFonts } from 'expo-font'
import LandingStyles from './LandingStyles'
import PropTypes from 'prop-types'
import Hero from '../../components/Hero/Hero'

export default function Landing ({ navigation }) {
  const [fontsLoaded] = useFonts({
    interExtraLight: require('../../assets/fonts/Inter-ExtraLight.ttf'),
    interBold: require('../../assets/fonts/Inter-Bold.ttf'),
    interSemiBold: require('../../assets/fonts/Inter-SemiBold.ttf'),
    barlowLight: require('../../assets/fonts/BarlowSemiCondensed-Light.ttf')
  })

  if (!fontsLoaded) {
    return <AppLoading />
  } else {
    return (
    <View style={LandingStyles.landingContainer}>
      <Hero />
      <View style={LandingStyles.landingMain}>
        <Text style={[{ fontFamily: 'interExtraLight' }, LandingStyles.subtitle]}>Welcome to</Text>
        <Text style={[{ fontFamily: 'barlowLight' }, LandingStyles.title]}>WAG</Text>
        <Text style={[{ fontFamily: 'interSemiBold' }, LandingStyles.description]}>The platform to store and share digital material! Do not lose anything!</Text>
        <Text style={[{ fontFamily: 'interSemiBold' }, LandingStyles.description]}>Join For Free</Text>

        <View style={LandingStyles.autenticationContainer}>
          <TouchableOpacity
            testID="login"
            onPress={() =>
              navigation.navigate('Login', { name: 'Login' })
            }
          >
            <Text style={[{ fontFamily: 'interBold' }, LandingStyles.loginText]}> I have an account </Text>
          </TouchableOpacity>
          <TouchableOpacity
            testID="register"
            style = {LandingStyles.register}
            onPress={() => navigation.navigate('Register', { name: 'Register' })}
          >
            <Text style = {LandingStyles.registerTetx}> Sing up </Text>
          </TouchableOpacity>
        </View>
      </View>

    </View>
    )
  }
}

Landing.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  })
}
