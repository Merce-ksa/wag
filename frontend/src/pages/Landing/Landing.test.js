import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react-native'
import Landing from './Landing'

describe('Given a Landing function', () => {
  describe('When is rendered', () => {
    test('Should Landing match the snapshot ', () => {
      const landing = renderer.create(<Landing />).toJSON()
      expect(landing).toMatchSnapshot()
    })
  })

  describe('When clicking login button', () => {
    test('Should calls the send handler', () => {
      const navigation = {
        navigate: jest.fn()
      }

      const { getByTestId } = render(<Landing navigation = { navigation } />)

      fireEvent.press(getByTestId('register'))
      expect(navigation.navigate).toHaveBeenCalled()
    })
    test('Should calls the send handler', () => {
      const navigation = {
        navigate: jest.fn()
      }

      const { getByTestId } = render(<Landing navigation = { navigation } />)

      fireEvent.press(getByTestId('login'))
      expect(navigation.navigate).toHaveBeenCalled()
    })
  })
})
