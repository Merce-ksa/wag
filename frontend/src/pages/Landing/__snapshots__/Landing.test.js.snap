// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Landing function When is rendered Should Landing match the snapshot  1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.5,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/wave.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/wag-logo.png",
          }
        }
        style={
          Object {
            "resizeMode": "contain",
            "width": "60%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "color": "black",
        "fontSize": 20,
        "padding": 5,
        "textAlign": "left",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "padding": 5,
          "textAlign": "left",
        }
      }
    >
      Welcome to
    </Text>
    <Text
      style={
        Object {
          "fontSize": 38,
          "padding": 5,
          "textAlign": "left",
        }
      }
    >
      WAG
    </Text>
    <Text
      style={
        Object {
          "color": "#7B7F9E",
          "fontSize": 14,
          "fontWeight": "bold",
          "padding": 5,
          "textAlign": "left",
        }
      }
    >
      The platform to store and share digital material! Do not lose anything!
    </Text>
    <Text
      style={
        Object {
          "color": "#7B7F9E",
          "fontSize": 14,
          "fontWeight": "bold",
          "padding": 5,
          "textAlign": "left",
        }
      }
    >
      Join For Free
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 10,
        "marginTop": 50,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="login"
    >
      <Text
        style={
          Object {
            "color": "#567DF4",
            "fontWeight": "bold",
          }
        }
      >
         I have an account 
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#567DF4",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 10,
          "opacity": 1,
        }
      }
      testID="register"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "padding": 20,
            "textAlign": "center",
          }
        }
      >
         Sing up 
      </Text>
    </View>
  </View>
</View>
`;

exports[`Given a Landing function When is rendered Should Landing match the snapshot  2`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.5,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/wave.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/wag-logo.png",
          }
        }
        style={
          Object {
            "resizeMode": "contain",
            "width": "60%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "color": "black",
        "fontSize": 20,
        "padding": 5,
        "textAlign": "left",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "padding": 5,
          "textAlign": "left",
        }
      }
    >
      Welcome to
    </Text>
    <Text
      style={
        Object {
          "fontSize": 38,
          "padding": 5,
          "textAlign": "left",
        }
      }
    >
      WAG
    </Text>
    <Text
      style={
        Object {
          "color": "#7B7F9E",
          "fontSize": 14,
          "fontWeight": "bold",
          "padding": 5,
          "textAlign": "left",
        }
      }
    >
      The platform to store and share digital material! Do not lose anything!
    </Text>
    <Text
      style={
        Object {
          "color": "#7B7F9E",
          "fontSize": 14,
          "fontWeight": "bold",
          "padding": 5,
          "textAlign": "left",
        }
      }
    >
      Join For Free
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 10,
        "marginTop": 50,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="login"
    >
      <Text
        style={
          Object {
            "color": "#567DF4",
            "fontWeight": "bold",
          }
        }
      >
         I have an account 
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#567DF4",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 10,
          "opacity": 1,
        }
      }
      testID="register"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "padding": 20,
            "textAlign": "center",
          }
        }
      >
         Sing up 
      </Text>
    </View>
  </View>
</View>
`;
